plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -1155,50 +1159,90 @@ public void renameTable(ConnectorSession session, ConnectorTableHandle tableHand
         catalog.renameTable(session, ((IcebergTableHandle) tableHandle).getSchemaTableName(), newTable);
     }
 
+    @SuppressWarnings("unchecked")
     @Override
     public void setTableProperties(ConnectorSession session, ConnectorTableHandle tableHandle, Map<String, Optional<Object>> properties)
     {
         IcebergTableHandle table = (IcebergTableHandle) tableHandle;
         BaseTable icebergTable = (BaseTable) catalog.loadTable(session, table.getSchemaTableName());
 
         transaction = icebergTable.newTransaction();
-        UpdateProperties updateProperties = transaction.updateProperties();
+        Optional<UpdateProperties> updateProperties = Optional.empty();
 
         for (Map.Entry<String, Optional<Object>> propertyEntry : properties.entrySet()) {
             String trinoPropertyName = propertyEntry.getKey();
             Optional<Object> propertyValue = propertyEntry.getValue();
 
             switch (trinoPropertyName) {
                 case FILE_FORMAT_PROPERTY:
-                    updateProperties.defaultFormat(((IcebergFileFormat) propertyValue.orElseThrow()).toIceberg());
+                    updateProperties = updateProperties.or(() -> Optional.of(transaction.updateProperties()))
+                                    .map(update -> update.defaultFormat(((IcebergFileFormat) propertyValue.orElseThrow()).toIceberg()));
                     break;
                 case FORMAT_VERSION_PROPERTY:
                     // UpdateProperties#commit will trigger any necessary metadata updates required for the new spec version
-                    updateProperty(updateProperties, FORMAT_VERSION, propertyValue, formatVersion -> Integer.toString((int) formatVersion));
+                    updateProperties = updateProperties.or(() -> Optional.of(transaction.updateProperties()))
+                            .map(update -> updateProperty(update, FORMAT_VERSION, propertyValue, formatVersion -> Integer.toString((int) formatVersion)));
```

Comment:
`.map` can be kind of misleading, since the value is always present at this point

```java
updateProperties = Optional.of(updateProperties.orElseGet(....)); // actually an ordinary `if` would be better
updateProperty(updateProperties.get() , ....);
  ```